Ensure old sphinx versions don't pull in incompatible docutils

dopplershift opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->

Right now this is allowing Sphinx 3.5.3 and docutils 0.18 to be installed together on Python 3.10--and it does NOT work.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Diff:
```
❯ ./show_diff.py
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
noarch::py-dom-xpath-six-0.2.3-pyhd8ed1ab_0.tar.bz2
-    "python =2.7|>=3.6"
+    "python ==2.7.*|>=3.6"
noarch::sphinx-3.0.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.3-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.4-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.1.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.1.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.1.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.2.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.2.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.3.0-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.3.1-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.3.1-pyhd8ed1ab_1.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.0-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.1-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.2-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.3-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.0-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.1-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.2-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.3-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
```

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

dopplershift commented:
Ok, updated to include sphinx 2.x since that would have then been the version installed on Python 3.10. Updated diff:
```
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
noarch::py-dom-xpath-six-0.2.3-pyhd8ed1ab_0.tar.bz2
-    "python =2.7|>=3.6"
+    "python ==2.7.*|>=3.6"
noarch::sphinx-2.0.0-py_1.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.0.0-py_2.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.0.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.1.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.1.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.1.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.2.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.2.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.2.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.3.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.3.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.4.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.4.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.4.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.4.3-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-2.4.4-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.3-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.0.4-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.1.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.1.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.1.2-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.2.0-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.2.1-py_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.3.0-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.3.1-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.3.1-pyhd8ed1ab_1.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.0-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.1-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.2-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.4.3-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.0-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.1-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.2-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
noarch::sphinx-3.5.3-pyhd8ed1ab_0.tar.bz2
-    "docutils >=0.12",
+    "docutils >=0.12,<0.17",
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
```

jakirkham commented:
Thanks Ryan! 😄
